### PR TITLE
deep immutable of array intersectsMeshes

### PR DESCRIPTION
**Pull request by nekochanoide**
array of deep immutable to deep immutable of array for Ray.prototype.intersectsMeshes
array is not being changed in method

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/13446/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13446/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13446/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/13446/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/13446/merge
https://gui.babylonjs.com/?snapshot=refs/pull/13446/merge
https://nme.babylonjs.com/?snapshot=refs/pull/13446/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/13446/merge#BCU1XR#0

File: packages/dev/core/src/Culling/ray.ts [hunk 1]
```diff
@@ -319,7 +319,7 @@ export class Ray {
      * @param results array to store result in
      * @returns Array of picking infos
      */
-    public intersectsMeshes(meshes: Array<DeepImmutable<AbstractMesh>>, fastCheck?: boolean, results?: Array<PickingInfo>): Array<PickingInfo> {
+    public intersectsMeshes(meshes: DeepImmutable<Array<AbstractMesh>>, fastCheck?: boolean, results?: Array<PickingInfo>): Array<PickingInfo> {
```
Comment (RaananW) [inline packages/dev/core/src/Culling/ray.ts:322]: The idea here is that the meshes are not being changed (i.e. immutable), the array is less of a concern. Is there a typescript-specific reason for that change?

**Comment by sebavan** [comment]
Closing for now, let s reopen if we better understand the need for this change ?

**Comment by nekochanoide** [comment]
@RaananW @sebavan 
I have this ts error:
```
Argument of type 'DeepImmutableArray<AbstractMesh>' is not assignable to parameter of type 'DeepImmutableObject<AbstractMesh>[]'.
  Type 'DeepImmutableArray<AbstractMesh>' is missing the following properties from type 'DeepImmutableObject<AbstractMesh>[]': pop, push, reverse, shift, and 6 more.ts(2345)
```
Meshes are not being changed. Array is not being changed. But TS is angry about types.

**Comment by RaananW** [comment]
> @RaananW @sebavan 
> I have this ts error:
> ```
> Argument of type 'DeepImmutableArray<AbstractMesh>' is not assignable to parameter of type 'DeepImmutableObject<AbstractMesh>[]'.
>   Type 'DeepImmutableArray<AbstractMesh>' is missing the following properties from type 'DeepImmutableObject<AbstractMesh>[]': pop, push, reverse, shift, and 6 more.ts(2345)
> ```
> Meshes are not being changed. Array is not being changed. But TS is angry about types.

Can you share the code that leads to this error? Maybe as a TS playground?

**Comment by nekochanoide** [comment]
@RaananW
https://playground.babylonjs.com/#WDZA5N#1

**Comment by sebavan** [comment]
Your code could be like this https://playground.babylonjs.com/#WDZA5N#2 ??? It simply ensures anything in the function will be unchanged at the mesh level ?

**Comment by RaananW** [comment]
> Your code could be like this https://playground.babylonjs.com/#WDZA5N#2 ??? It simply ensures anything in the function will be unchanged at the mesh level ?

TBH I am not sure why the object type is not DeepImmutableArray<AbstractMesh>.

**Pull request closed without merging**